### PR TITLE
Added a Way to select the number of columns for a image in FO

### PR DESCRIPTION
**Pull request by jdperea**
image

Please give me a feed back to make my code better.
I have to set the values (width and height) of images to 0 to use 100% of div

image

### CODE REVIEW

File: views/templates/hook/hook.tpl [hunk 1]
```diff
@@ -29,7 +29,7 @@
 			{if $hook == 'left' || $hook == 'right'}
 				<li class="htmlcontent-item-{$smarty.foreach.items.iteration|escape:'htmlall':'UTF-8'} col-xs-12">
 			{else}
-				<li class="htmlcontent-item-{$smarty.foreach.items.iteration|escape:'htmlall':'UTF-8'} col-xs-4">
+				<li class="htmlcontent-item-{$smarty.foreach.items.iteration|escape:'htmlall':'UTF-8'} col-xs-4  {$hItem.columns} {if $hItem.columns > 1}item-{$hItem.columns}-col{/if}">
```
Comment (Progi1984) [inline views/templates/hook/hook.tpl:32]:
```suggestion
				<li class="htmlcontent-item-{$smarty.foreach.items.iteration|escape:'htmlall':'UTF-8'} col-xs-4{$hItem.columns} {if $hItem.columns > 1}item-{$hItem.columns}-col{/if}">
```

File: themeconfigurator.php [hunk 2]
```diff
@@ -577,7 +583,7 @@ protected function addItem()
 
 		if (!Db::getInstance()->Execute('
 			INSERT INTO `'._DB_PREFIX_.'themeconfigurator` (
-					`id_shop`, `id_lang`, `item_order`, `title`, `title_use`, `hook`, `url`, `target`, `image`, `image_w`, `image_h`, `html`, `active`
+					`id_shop`, `id_lang`, `item_order`, `title`, `title_use`, `hook`, `url`, `target`, `image`, `image_w`, `image_h`, `html`,`columns`, `active`
```
Comment (Progi1984) [inline themeconfigurator.php:586]:
```suggestion
					`id_shop`, `id_lang`, `item_order`, `title`, `title_use`, `hook`, `url`, `target`, `image`, `image_w`, `image_h`, `html`, `columns`, `active`
```

File: themeconfigurator.php [hunk 3]
```diff
@@ -418,6 +420,9 @@ protected function updateItem()
 		$title = Tools::getValue('item_title');
 		$content = Tools::getValue('item_html');
 
+        $columns = (int)Tools::getValue('item_columns');
```
Comment (Progi1984) [inline themeconfigurator.php:423]:
```suggestion
		$content = Tools::getValue('item_html');
        	$columns = (int)Tools::getValue('item_columns');
```

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Thank you for your contribution @jdperea.

Some improvements : 
- if columns are defined, please hide width & height input boxes 
- add a sql migration script for update

**Comment by matks** [comment]
Hi and thanks for your contribution.

This repository is now deprecated. Sorry for the late answer 😞 

Kind regards

**Pull request closed without merging**